content/en/platform/corda/5.0/developing-applications/basic-cordapp/state.md

```diff
@@ -0,0 +1,195 @@
+---
+date: '2023-05-03'
+title: "Write States"
+version: 'Corda 5.0'
+menu:
+  corda5:
+    identifier: corda5-develop-first-cordapp-state
+    parent: corda5-develop-first-cordapp
+    weight: 2000
+---
+
+# Write States
+
+This tutorial guides you through writing the two states you need in your CorDapp: `AppleStamp` and `BasketofApples`.
+You will create these states in the `contracts/src/main/kotlin/com/r3/developers/apples/states` directory.
+
+## Learning Objectives
+
+After you have completed this tutorial, you will know how to create and implement states in a CorDapp.
+
+## Before You Start
+
+Before you start building states, read more about [states]({{< relref "../components/ledger/states.md" >}}).
+
+
+## Create the `AppleStamp` State
+
+First, create the `AppleStamp` state. This state is the voucher issued to customers.
+
+1. Go to the `contracts/src/main/kotlin/com/r3/developers/apples` folder.
+2. Right-click the **states** folder, select **New > Kotlin Class** and create a file called `AppleStamp`.
+3. Open the file.
+
+### Add Annotations
+
+The first thing you should do when writing a state is add the `@BelongsToContract` annotation. This annotation
+establishes the relationship between a state and a contract. Without this, your state does not know which contract is used to verify it.
+
+Add the annotation `@BelongsToContract(AppleStampContract::class)` to your state. The contract will be defined later.
+
+This what your code should look like so far:
+
+```kotlin
+@BelongsToContract(AppleStampContract.class)
+```
+
+{{< note >}}
+Adding this annotation triggers an error in IntelliJ because you have not created the `AppleStampContract` yet. Ignore this error for now - you will add the contract class in the [Write Contracts]({{< relref "./contract.md" >}}) tutorial.
+{{< /note >}}
+
+When naming your CorDapp files, it's best practice to match your contract and state names. In this case the state is called `AppleStamp`, so the contract is called `AppleStampContract`. Follow this naming convention when you write your own original CorDapp to avoid confusion.
```

Comment:
```suggestion
When naming your CorDapp files, it is best practice to match your contract and state names. In this case the state is called `AppleStamp`, so the contract is called `AppleStampContract`. Follow this naming convention when you write your own original CorDapp to avoid confusion.
```